src/main/java/org/broadinstitute/hellbender/utils/smithwaterman/SmithWatermanAligner.java

```diff
@@ -0,0 +1,34 @@
+package org.broadinstitute.hellbender.utils.smithwaterman;
+
+import org.broadinstitute.gatk.nativebindings.smithwaterman.SWOverhangStrategy;
+import org.broadinstitute.gatk.nativebindings.smithwaterman.SWParameters;
+
+import java.io.Closeable;
+import java.util.function.Supplier;
+
+public interface SmithWatermanAligner extends Closeable{
```

Comment:
This interface is defined here because the return type is dependent onhtsjdk classes, specifically CIGAR.  We wanted to keep any dependencies out of gatk-native-bindings so that we don't have yet another triangular htsjdk dependency.  I debated having just the `SWNativeBindings` interface and having to have callers translate from String -> cigar but I think this is probably better.

Reply:
Make sense...